dev/NavigationView/NavigationView.cpp

```diff
@@ -1530,8 +1512,36 @@ void NavigationView::UpdatePaneLayout()
                         // We know the actual height of footer items, so use that to determine how to split pane.
                         if (const auto& menuItems = m_leftNavRepeater.get())
                         {
-                            const auto footersActualHeight = footerItemsRepeater.ActualHeight();
-                            const auto menuItemsActualHeight = menuItems.ActualHeight();
+                            const auto footersActualHeight = [this, footerItemsRepeater]() {
+                                const auto footerItemsRepeaterMargin = footerItemsRepeater.Margin();
+                                return footerItemsRepeater.ActualHeight() + footerItemsRepeaterMargin.Top + footerItemsRepeaterMargin.Bottom;
+                            }();
+                            
+                            const auto menuItemsActualHeight = [this, menuItems]() {
+                                const auto menuItemsMargin = menuItems.Margin();
+                                return menuItems.ActualHeight() + menuItemsMargin.Top + menuItemsMargin.Bottom;
+                            }();
+
+                            const auto paneFooterActualHeight = [this]() {
+                                if (const auto& paneFooter = m_leftNavFooterContentBorder.get())
+                                {
+                                    const auto paneFooterMargin = paneFooter.Margin();
+                                    return paneFooter.ActualHeight() + paneFooterMargin.Top + paneFooterMargin.Bottom;
+                                }
+                                return 0.0;
+                            }();
+
+                            const auto visualItemsSeparatorHeight = [this]() {
+                                if (const auto& visualItemsSeparator = m_visualItemsSeparator.get())
+                                {
+                                    const auto visualItemsSeparatorMargin = visualItemsSeparator.Margin();
+                                    return visualItemsSeparator.ActualHeight() + visualItemsSeparatorMargin.Top + visualItemsSeparatorMargin.Bottom;
+                                }
+                                return 0.0;
+                            }();
+
+                            // Footer, PaneFooter, and VisualItemsSeparator are included in the footerGroup to calculate available height for menu items.
+                            const auto footerGroupActualHeight = footersActualHeight + paneFooterActualHeight + visualItemsSeparatorHeight;
```

Comment:
Adding the visualItemsSeparatorHeight feels incorrect to me:  you don't want the separator to be shown because of a lack of space due to the separator being shown. You know what I mean?

Reply:
I thought about that too... but it was part of the calculation before so I simply included it.

Reply:
Discussed offline, will remove.